packages/ai-anthropic/src/node/anthropic-language-model.ts

```diff
@@ -191,7 +191,11 @@ export class AnthropicModel implements LanguageModel {
                         }
                         if (contentBlock.type === 'tool_use') {
                             toolCall = { name: contentBlock.name!, args: '', id: contentBlock.id!, index: event.index };
-                            yield { tool_calls: [{ finished: false, id: toolCall.id, function: { name: toolCall.name, arguments: toolCall.args } }] };
+                            const sticky = (() => {
+                                const found = request.tools?.find(t => t.name === (toolCall as { name?: string }).name);
+                                return found?.sticky ?? 'args';
+                            })();
+                            yield { tool_calls: [{ finished: false, id: toolCall.id, function: { name: toolCall.name, arguments: toolCall.args }, sticky }] };
```

Comment:
I did not expect the "sticky" being part of the returned tool call. I assumed we implement this feature by checking the `sticky` flag of the tool request when mapping the Theia messages to the respective provider messages.

Thinking about it, I guess it makes sense to "raise" the level of the stickiness feature and bringing it into the chat model. This way the actual agents can decide on what to do with the tool calls.